datastore/query.js

```diff
@@ -0,0 +1,388 @@
+
+  // [START anscestor_query]
+  // SELECT * FROM Task WHERE __key__ HAS  ANCESTOR ("TaskList", "default")
+  var ancestorKey = datastore.key(['TaskList', 'default']);
+
+  var query = datastore.createQuery('Task')
+    .hasAncestor(ancestorKey);
+  // [END anscestor_query]
+
+  return query;
+};
+
+Query.prototype.testRunQuery = function(callback) {
+  var query = this.basicQuery;
+
+  // jshint unused:false
+  // [START run_query]
+  datastore.runQuery(query, function(err, entities) {
+    if (!err) {
+      // Entities found.
+    }
+  });
+  // [END run_query]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testPropertyFilter = function(callback) {
+  var datastore = this.datastore;
+
+  // [START property_filter]
+  var query = datastore.createQuery('Task')
+    .filter('done =', false);
+  // [END property_filter]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testCompositeFilter = function(callback) {
+  var datastore = this.datastore;
+
+  // [START composite_filter]
+  var query = datastore.createQuery('Task')
+    .filter('done =', false)
+    .filter('priority =', 4);
+  // [END composite_filter]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testKeyFilter = function(callback) {
+  var datastore = this.datastore;
+
+  // [START key_filter]
+  var query = datastore.createQuery('Task')
+    .filter('__key__ =', datastore.key(['Task', 'someTask']));
+  // [END key_filter]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testAscendingSort = function(callback) {
+  var datastore = this.datastore;
+
+  // [START ascending_sort]
+  var query = datastore.createQuery('Task')
+    .order('created');
+  // [END ascending_sort]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testDescendingSort = function(callback) {
+  var datastore = this.datastore;
+
+  // [START descending_sort]
+  var query = datastore.createQuery('Task')
+    .order('-created');
+  // [END descending_sort]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testMultiSort = function(callback) {
+  var datastore = this.datastore;
+
+  // [START multi_sort]
+  var query = datastore.createQuery('Task')
+    .order('-priority')
+    .order('created');
+  // [END multi_sort]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testKindlessQuery = function(callback) {
+  var datastore = this.datastore;
+  var lastSeenKey = this.datastore.key(['Task', Date.now()]);
+
+  // [START kindless_query]
+  var query = datastore.createQuery()
+    .filter('__key__ =', lastSeenKey);
+  // [END kindless_query]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testRunQueryProjection = function(callback) {
+  var query = this.projectionQuery;
+
+  // jshint unused:false
+  // [START run_query_projection]
+  datastore.runQuery(query, function(err, entities) {
+    if (!err) {
+      // Entities found.
```

Comment:
Please build up the priorities and percent_completes list. This is to show how to access the properties in a projection.